components/components-azurestorage/src/main/java/org/talend/components/azurestorage/tazurestorageconnection/TAzureStorageConnectionDefinition.java

```
@@ -0,0 +1,64 @@
+//
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.components.azurestorage.tazurestorageconnection;
+
+import org.talend.components.api.component.ConnectorTopology;
+import org.talend.components.api.component.runtime.ExecutionEngine;
+import org.talend.components.api.properties.ComponentProperties;
+import org.talend.components.azurestorage.AzureStorageDefinition;
+import org.talend.components.azurestorage.blob.runtime.AzureStorageSourceOrSink;
+import org.talend.daikon.properties.property.Property;
+import org.talend.daikon.runtime.RuntimeInfo;
+
+/**
+ * @author undx
+ *
+ */
+public class TAzureStorageConnectionDefinition extends AzureStorageDefinition {
+
+    public static final String COMPONENT_NAME = "tAzureStorageConnection"; //$NON-NLS-1$
+
+    public TAzureStorageConnectionDefinition() {
+        super(COMPONENT_NAME);
+    }
+
+    @Override
+    public Class<? extends ComponentProperties> getPropertyClass() {
+        return TAzureStorageConnectionProperties.class;
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
+    public Class getPropertiesClass() {
+        return TAzureStorageConnectionProperties.class;
+    }
+
+    /**
+     * FIXME - Currently we don't return this values ! Have to create a source and sink class for this ?
+     */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public Property[] getReturnProperties() {
+        return new Property[] { RETURN_ERROR_MESSAGE_PROP, RETURN_ACCOUNT_NAME_PROP, RETURN_ACCOUNT_KEY_PROP };
+    }
+
+    @Override
+    public RuntimeInfo getRuntimeInfo(ExecutionEngine engine, ComponentProperties properties,
+            ConnectorTopology connectorTopology) {
+        if (connectorTopology == ConnectorTopology.NONE) {
```

Recent update:
This method should not return null. It should throw exception in case of unsupported engine or connectorTopology is passed.
You may use 
assertConnectorTopologyCompatibility() 
assertEngineCompatibility()
methods from AbstractComponentDefinition
to check incoming arguments. Then simply return RuntimeInfo (without any if-else)